azure-mgmt-redis/src/main/java/com/microsoft/azure/management/redis/RedisCache.java

```diff
@@ -0,0 +1,497 @@
+             * @return the next stage of Redis Cache definition.
+             */
+            WithCreate withBasicSku(int capacity);
+
+            /**
+             * Specifies the Standard Sku of the Redis Cache.
+             *
+             * @return the next stage of Redis Cache definition.
+             */
+            WithCreate withStandardSku();
+
+            /**
+             * Specifies the Standard sku of the Redis Cache.
+             *
+             * @param capacity specifies what size of Redis Cache to deploy for Standard sku with C family (0, 1, 2, 3, 4, 5, 6).
+             * @return the next stage of Redis Cache definition.
+             */
+            WithCreate withStandardSku(int capacity);
+
+            /**
+             * Specifies the Premium sku of the Redis Cache.
+             *
+             * @return the next stage of Redis Cache definition.
+             */
+            WithPremiumSkuCreate withPremiumSku();
+
+            /**
+             * Specifies the Premium sku of the Redis Cache.
+             *
+             * @param capacity specifies what size of Redis Cache to deploy for Standard sku with P family (1, 2, 3, 4).
+             * @return the next stage of Redis Cache definition.
+             */
+            WithPremiumSkuCreate withPremiumSku(int capacity);
+        }
+
+         /**
+         * A Redis Cache definition with sufficient inputs to create a new
+         * Redis Cache in the cloud, but exposing additional optional inputs to
+         * specify.
+         */
+        interface WithCreate extends
+                 Creatable<RedisCache>,
+                 DefinitionWithTags<WithCreate> {
+             /**
+              * Enables non-ssl Redis server port (6379).
+              *
+              * @return the next stage of Redis Cache definition.
+              */
+             WithCreate withNonSslPortEnabled();
+
+             /**
+              * Disables non-ssl Redis server port (6379).
+              *
+              * @return the next stage of Redis Cache definition.
+              */
+             WithCreate withNonSslPortDisabled();
+
+             /**
+              * All Redis Settings. Few possible keys:
+              * rdb-backup-enabled, rdb-storage-connection-string, rdb-backup-frequency, maxmemory-delta, maxmemory-policy,
+              * notify-keyspace-events, maxmemory-samples, slowlog-log-slower-than, slowlog-max-len, list-max-ziplist-entries,
+              * list-max-ziplist-value, hash-max-ziplist-entries, hash-max-ziplist-value, set -max-intset-entries,
+              * zset-max-ziplist-entries, zset-max-ziplist-value etc.
+              *
+              * @param redisConfiguration configuration of Redis Cache as a map indexed by configuration name
+              * @return the next stage of Redis Cache definition.
+              */
+             WithCreate withRedisConfiguration(Map<String, String> redisConfiguration);
```

Comment:
if there is a specific set of strings that this method recognizes, shouldn't the key be an enum defining those possible values?  Or are arbitrary strings supported?